spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/metadata/JdbcMetadataStore.java

```diff
@@ -49,32 +49,24 @@
 
 	private final JdbcOperations jdbcTemplate;
 
-	private volatile String tablePrefix = DEFAULT_TABLE_PREFIX;
+	private String tablePrefix = DEFAULT_TABLE_PREFIX;
 
-	private volatile String region = "DEFAULT";
+	private String region = "DEFAULT";
 
-	private String getValueQuery = "SELECT METADATA_VALUE FROM %SMETADATA_STORE WHERE METADATA_KEY=? AND REGION=?";
+	private String lockHint = "FOR UPDATE";
 
-	private String getValueForUpdateQuery = "SELECT METADATA_VALUE FROM %SMETADATA_STORE WHERE METADATA_KEY=? AND REGION=? FOR UPDATE";
+	private String getValueQuery = "SELECT METADATA_VALUE FROM %sMETADATA_STORE WHERE METADATA_KEY=? AND REGION=?";
 
-	private String replaceValueQuery = "UPDATE %SMETADATA_STORE SET METADATA_VALUE=? WHERE METADATA_KEY=? AND METADATA_VALUE=? AND REGION=?";
+	private String getValueForUpdateQuery = "SELECT METADATA_VALUE FROM %sMETADATA_STORE WHERE METADATA_KEY=? AND REGION=? %s";
```

Comment:
Is it safe to assume that the "hint" always follows the query?

Maybe some vendor has `FOR UPDATE SELECT...` ??

Reply:
I check Hibernate's `Dialect` hierarchy. It does this in the `SimpleSelect`:
```
buf.append( " from " )
				.append( dialect.appendLockHint( lockOptions, tableName ) );
```
Which ends up with the (`SQLServerDialect`):
```
return tableName + " with (updlock, rowlock)";
```
Or there is this:
```
public String applyLocksToSql(String sql, LockOptions aliasedLockOptions, Map<String, String[]> keyColumnNames) {
		return sql + new ForUpdateFragment( this, aliasedLockOptions, keyColumnNames ).toFragmentString();
	}
```

So, looks like it always comes in the end.

But I won't mind if someone may provide some other hints to revise it.